src/ophyd_async/epics/eiger/_eiger_io.py

```diff
@@ -0,0 +1,58 @@
+from enum import Enum
+
+from ophyd_async.core import AsyncStatus, Device
+from ophyd_async.epics.signal import epics_signal_r, epics_signal_rw_rbv, epics_signal_w
+
+
+class EigerTriggerMode(str, Enum):
+    internal = "ints"
+    edge = "exts"
+    gate = "exte"
+
+
+class PhotonEnergy(Device):
+    """Changing photon energy takes some time so only do so if the current energy is
+    outside the tolerance."""
+
+    def __init__(self, prefix, tolerance=0.1, name: str = "") -> None:
+        self._photon_energy = epics_signal_rw_rbv(float, f"{prefix}PhotonEnergy")
+        self.tolerance = tolerance
+        super().__init__(name)
+
+    @AsyncStatus.wrap
+    async def set(self, value):
+        current_energy = await self._photon_energy.get_value()
+        if abs(current_energy - value) > self.tolerance:
+            await self._photon_energy.set(value)
```

Comment:
Slightly concerned about putting this logic in the IO class. As written this will break load/save as it's not a Signal.

What's the use case for setting within a tolerance? If it's from a custom plan, could the logic be lifted into a plan stub? If not would a `set_energy` method on the detector be acceptable?

Reply:
The use-case is that we want to avoid setting it as much as possible because it increases the time it takes to arm the detector and it's scientifically fine for it to be just within the tolerance. At the moment we bundle all the logic required to set up and arm the detector into a single `bps.set` because we need to do it all in parallel to another plan running. If we want to do this via `plan_stubs` then I guess we're blocked on https://github.com/bluesky/bluesky/issues/1652. We could put it in a `set_energy` and call it during the `arm`

Reply:
As discussed in person, we put this in `arm`.

Reply:
I have put it in `prepare`. I would rather put it in the `arm` so that we have a better ordering of when to do it but to do so would need to change the `arm` interface quite a bit, probably to take something that inherits from `TriggerInfo`. Are you happy for me to do this refactor?

Reply:
@ZohebShaikh will be doing this as part of https://github.com/bluesky/ophyd-async/issues/405, so it would be useful if you could review and/or add to his PR to make sure the Eiger usecase is covered